### PR TITLE
check passwords are enforced when creating new public share

### PR DESCRIPTION
**Pull request by individual-it**
## Description
check that the password enforcement works correctly when creating new public link

## Related Issue
part of  #2075

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = {
 
       return this.click({ locateStrategy: 'xpath', selector: createLinkButtonXpath })
         .waitForOutstandingAjaxCalls()
-        .waitForElementNotPresent({ locateStrategy: 'xpath', selector: createLinkButtonXpath })
+        .waitForElementNotPresent({ locateStrategy: 'xpath', selector: createLinkButtonXpath, abortOnFailure: false })
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: do not fail the function when the button is not gone in time, that way the same function can be used for steps that expect the sharing not to work

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 2]
```diff
@@ -655,3 +655,8 @@ Then('the file {string} shared by {string} should not be in {string} state', fun
 Then('file/folder {string} should be marked as shared by {string} on the webUI', function (element, sharer) {
   return client.page.sharedWithMePage().assertSharedByUser(element, sharer)
 })
+
+Then('the user {string} should not have created any shares', async function (user) {
+  const shares = await sharingHelper.getAllSharesSharedByUser(user)
+  assert.deepStrictEqual(shares, [], 'There should not be any share, but there are')
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Better to test for length than the content.
```suggestion
  assert.strictEqual(shares.length, 0, 'There should not be any share, but there are')
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 3]
```diff
@@ -150,8 +156,14 @@ module.exports = {
         return res.ocs.data
       })
   },
+  getAllSharesSharedWithUser: function (user) {
+    return this.getAllShares(user, true)
+  },
+  getAllSharesSharedByUser: function (user) {
+    return this.getAllShares(user, false)
```
Comment (skshetry) [inline tests/acceptance/helpers/sharingHelper.js]: Default is false, so saving a few characters.
```suggestion
    return this.getAllShares(user)
```

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 4]
```diff
@@ -43,6 +43,21 @@ module.exports = {
...
+    },
+    /**
+     *
+     * @returns {Promise<string>}
+     */
+    getErrorMessage: async function () {
+      let result
+      await this.getText(
+        'xpath',
+        this.elements.publicLinkContainer.selector + this.elements.errorMessageInsidePublicLinkContainer.selector,
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: Would be good to keep this in a variable inside this function.

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 5]
```diff
@@ -43,6 +43,21 @@ module.exports = {
...
+        function (textValue) {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: Generally, the convention is to name this `result`.
```suggestion
        function (result) {
```

Or, just destructure to get the value:
```suggestion
        function ({value}) {
```

**Comment by individual-it** [comment]
@skshetry addessed your comments

**Pull request merged**